### DISCUSSION TITLE
Multi-currency

### DISCUSSION THREAD

**Discussion by jramoyo-serenade** [swellstores/community#513] [Help] [answered]
We've just enabled our production store, and I am investigating an issue where products created with non-USD prices are not displaying any prices in the dashboard (even `GET /products/{id}` return `"price": null`).

Note that we are still able to create orders against these products and total prices are calculating correctly.

To replicate, I created a new test store and I can see that by default, products with non-USD prices display as expected in the dashboard:
![1](https://github.com/swellstores/community/assets/133632539/021da427-8b92-4fff-b311-489cc003017e)

However, as soon as I enabled multi-currency in "General" -> "Settings" (**like what we have in production**), the prices no longer appear in the dashboard:
![2](https://github.com/swellstores/community/assets/133632539/fad80a9e-6664-490c-b204-e301f464cc38)

Even if I add the currencies, they still don't show-up:
![3](https://github.com/swellstores/community/assets/133632539/e2381651-03f7-4266-bb87-fe60693ebebe)

I found out that if I temporarily set a currency as default...
![4](https://github.com/swellstores/community/assets/133632539/98e3c9a6-ac9a-4d41-8fa8-20f1bc1462b3)

And switch the default currency back to USD, the previously default currency becomes "Priced":
![5](https://github.com/swellstores/community/assets/133632539/428baf60-90c9-49bd-b781-117dc555aff4)

Then the currency displays in the dashboard:
![6](https://github.com/swellstores/community/assets/133632539/730b3cfc-0a0e-4f7f-aa8e-4696aed3098b)

In fact, if I do this for all currencies, I can manage to make them all show-up in the dashboard:
![7](https://github.com/swellstores/community/assets/133632539/176b3638-9575-4638-9b69-d4e9a84d447b)

Having said that, I have some questions...

1. What is the expected behaviour? Why does enabling multi-currency hide the prices for non-USD products.
2. Once enabled, how can I disable multi-currency? There isn't an option to revert back. Given that the default behaviour achieves what we expect, I'd rather just disable multi-currency.
2. Is it safe to cycle through default currencies like how I demonstrated above?

**Comment by jramoyo-serenade** [comment]
Any ideas @logeshswell @ericingram?

**Reply by logeshswell** [reply]
@jramoyo-serenade Let me take a deeper look and update you shortly.

**Comment by surenw-serenade** [comment]
@logeshswell any updates on this?

**Comment by logeshswell** [comment]
> However, as soon as I enabled multi-currency in "General" -> "Settings" (like what we have in production), the prices no longer appear in the dashboard:
> What is the expected behavior? Why does enabling multi-currency hide the prices for non-USD products.

To answer your question, before enabling multi-currency, the price is stored in `product.purchase_options.standard.price` field.
But after enabling multi-currency, you could check the price value in `product.purchase_options.standard.$currency.USD.price` field

**Comment by logeshswell** [comment]
> However, as soon as I enabled multi-currency, the prices no longer appear in the dashboard:

This is possible if the `product.currency` value is a **Display** instead of **Priced** type. 
That's what I see from the screenshot you shared below. Only USD is **Base** (or priced), and the other currencies are defined as **Display** type.

![311583024-e2381651-03f7-4266-bb87-fe60693ebebe](https://github.com/swellstores/community/assets/100766131/55bda886-4e7b-4c05-8198-9bfec8a59690)

**Reply by logeshswell** [reply, marked as answer]
If the product currency is set to a **Display** currency, then Swell admin dashboard won't display the value which is an expected behavior.

However, you can still create orders against these products, and the total prices will be calculated properly based on the present currency exchange rate.

**Comment by logeshswell** [comment]
> Once enabled, how can I disable multi-currency?

You should first delete all currencies. Then, edit the base currency to disable multi-currency as follows:

<img width="679" alt="Screenshot 2024-04-04 at 1 47 42 PM" src="https://github.com/swellstores/community/assets/100766131/10b43cd4-e454-4ba3-bb57-4c2b5ec7c646">

**Comment by logeshswell** [comment]
> Is it safe to cycle through default currencies like how I demonstrated above?

It totally depends on your use case and how you fetch the product pricing. But, it is completely safe to demonstrate in local env, and you won't experience data losses.
